charts/fluentd/templates/daemonset.yaml

```diff
@@ -31,6 +31,26 @@ spec:
               secretKeyRef:
                 name: {{ template "fullname" . }}
                 key: elasticsearch_port
+          - name:  FLUENT_ELASTICSEARCH_SCHEME
+            valueFrom:
+              secretKeyRef:
+                name: {{ template "fullname" . }}
+                key: elasticsearch_scheme
+          {{ if .Values.elasticsearch.username }}
```

Comment:
Having had a look at this it seems like there's no easy way to handle optional values in Secrets - because values must be b64 encoded, null values cause syntax errors, so it seems like wrapping them in `{{ if }}` blocks is the simplest option.